src/robotFactory.js

```diff
@@ -1,10 +1,85 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    const x = coords.x ? coords.x : 0;
```

Comment:
```suggestion
    const x = coords.x || 0;
```
it will be the same, but clearer